tools/pepsi/src/aliveness.rs

```diff
@@ -0,0 +1,162 @@
+use std::{collections::BTreeMap, net::IpAddr, time::Duration};
+
+use aliveness_client::{Aliveness, AlivenessState, ServiceState, SystemServices};
+use clap::{arg, Args};
+use color_eyre::Result;
+use colored::Colorize;
+use repository::SDK_VERSION;
+
+use crate::parsers::NaoAddress;
+
+#[derive(Args)]
+pub struct Arguments {
+    /// Output verbose version of the aliveness information
+    #[arg(long, short = 'v')]
+    verbose: bool,
+    /// Output aliveness information as json
+    #[arg(long, short = 'j')]
+    json: bool,
+    /// Timeout in ms for waiting for responses
+    #[arg(long, short = 't', default_value = "200")]
+    timeout: u64,
+    /// The NAOs to show the aliveness information from, e.g. 20w or 10.1.24.22
+    naos: Option<Vec<NaoAddress>>,
+}
+
+type AlivenessList = BTreeMap<IpAddr, AlivenessState>;
+
+pub async fn aliveness(arguments: Arguments) -> Result<()> {
+    let states = query_aliveness(&arguments).await?;
+    if arguments.json {
+        println!("{}", serde_json::to_string(&states)?);
+    } else if arguments.verbose {
+        print_verbose(&states);
+    } else {
+        print_summary(&states);
+    }
+    Ok(())
+}
+
+fn print_summary(states: &AlivenessList) {
+    const SPACING: usize = 3;
+    const CHARGE_OK_THRESHOLD: f32 = 0.95;
+    const CHARGING_ICON: &str = "󱐋";
+    const DISCHARGING_ICON: &str = "󰁽";
+    const OS_ICON: &str = "󱑞";
+    const ALL_OK_ICON: &str = "✔";
+
+    for (ip, state) in states.iter() {
+        let id = match ip {
+            IpAddr::V4(ip) => ip.octets()[3],
+            IpAddr::V6(ip) => ip.octets()[15],
+        };
+
+        let mut output = String::new();
+
+        if let Some(battery) = state.battery {
+            if battery.charge < CHARGE_OK_THRESHOLD {
+                let charge = (battery.charge * 100.0) as u32;
+                let icon = if battery.current >= 0.0 {
+                    CHARGING_ICON
+                } else {
+                    DISCHARGING_ICON
+                };
+
+                output.push_str(&format!("{icon} {charge}%{:SPACING$}", ""))
```

Comment:
What do you think of tabs (`\t`) vs the spacing?

Reply:
Tabs may lead to highly inconsistent spacing depending on position and length of the previous entry (and probably also terminal emulator settings). While there may be an advantage that some cells can become column-wise aligned, cells could also be misaligned entirely. Overall I think the approach of using a fixed spacing leads to more consistent results.